### PR TITLE
Update vendor k8s 1.26

### PR DESCRIPTION
**Pull request by kushagra98**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR updates VPA and VPA e2e dependencies for VPA and VPA-e2e tests to K8s v1.26.

Updated VPA vendor to 1.26, fixed files with outdated things.
Updated VPA-e2e vendor to 1.26, fixed files with outdated things. (https://github.com/kubernetes/kubernetes/commit/dfdf88d4faafa6fd39988832ea0ef6d668f490e9)
Added script to update VPA vendor (equivalent to https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/hack/update-kubernetes-deps-in-e2e.sh)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by kushagra98** [comment]
OOM tests completes in ~ 18 min. Increased the time limit to 20min.
I've run full-vpa tests with these changes (both VPA and tests) and they passed.

**Comment by kushagra98** [comment]
@jbartosik this PR is ready for review. Addressed comments from #5527.

File: vertical-pod-autoscaler/e2e/v1/full_vpa.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ const (
 	// the initial values should be outside minimal bounds
 	initialCPU     = int64(10) // mCPU
 	initialMemory  = int64(10) // MB
-	oomTestTimeout = 12 * time.Minute
+	oomTestTimeout = 20 * time.Minute
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1/full_vpa.go:46]: This change worries me - the point of OOM test is that it should be quicker than scaling without OOMs.

With this we expect them to be slower. `pollTmeout` is 15 minuutes ([v1beta2](https://github.com/kubernetes/autoscaler/blob/04e5bfc88363b4af9fdeb9dfd06c362ec5831f51/vertical-pod-autoscaler/e2e/v1beta2/common.go#L51), [v1](https://github.com/kubernetes/autoscaler/blob/04e5bfc88363b4af9fdeb9dfd06c362ec5831f51/vertical-pod-autoscaler/e2e/v1/common.go#L51)).

Can you explain why we need to increase this timeout?

**Comment by kushagra98** [comment]
Reverted the timeout to 12s. Seems like I was using a cluster with small node side. VPA couldn't evict pod. The OOM test takes ~ 7min.

File: vertical-pod-autoscaler/e2e/v1/full_vpa.go (hunk 1 above)
Comment (kushagra98) [reply vertical-pod-autoscaler/e2e/v1/full_vpa.go]: Reverted back. Thanks!

**Review by jbartosik — APPROVED** [review]
/lgtm
/approve

**Pull request merged**